ndc-reference/bin/reference/main.rs

```diff
@@ -1541,19 +1589,100 @@ fn eval_dimension(
             arguments,
             field_path,
             path,
-        } => eval_column_at_path(
-            collection_relationships,
-            variables,
-            state,
-            row,
-            path,
-            column_name,
-            arguments,
-            field_path.as_deref(),
-        ),
+            extraction,
+        } => {
+            let value = eval_column_at_path(
+                collection_relationships,
+                variables,
+                state,
+                row,
+                path,
+                column_name,
+                arguments,
+                field_path.as_deref(),
+            )?;
+
+            eval_extraction(extraction, value)
+        }
     }
 }
 // ANCHOR_END: eval_dimension
+// ANCHOR: eval_extraction
+fn eval_extraction(
+    extraction: &Option<ndc_models::ExtractionFunctionName>,
+    value: serde_json::Value,
+) -> Result<serde_json::Value> {
+    match extraction {
+        None => Ok(value),
+        Some(extraction) => {
+            let iso8601::Date::YMD { year, month, day } = iso8601::date(value.as_str().ok_or({
+                (
+                    StatusCode::INTERNAL_SERVER_ERROR,
+                    Json(ndc_models::ErrorResponse {
+                        details: serde_json::Value::Null,
+                        message: "Expected date".into(),
+                    }),
+                )
+            })?)
+            .map_err(|_| {
+                (
+                    StatusCode::INTERNAL_SERVER_ERROR,
+                    Json(ndc_models::ErrorResponse {
+                        details: serde_json::Value::Null,
+                        message: "Unable to parse date".into(),
+                    }),
+                )
+            })?
+            else {
+                return Err((
+                    StatusCode::INTERNAL_SERVER_ERROR,
+                    Json(ndc_models::ErrorResponse {
+                        details: serde_json::Value::Null,
+                        message: "Invalid date format".into(),
+                    }),
+                ));
+            };
+
+            match extraction.as_str() {
+                "year" => serde_json::to_value(year).map_err(|_| {
+                    (
+                        StatusCode::INTERNAL_SERVER_ERROR,
+                        Json(ndc_models::ErrorResponse {
+                            details: serde_json::Value::Null,
+                            message: "Cannot encode date part".into(),
```

Comment:
```suggestion
                            message: "Cannot encode date year part".into(),
```

Different errors with identical strings are incredible irritating to debug.